### ISSUE TITLE
Logo renders for a split second at top left of screen before parent loads.

### ISSUE THREAD

**Issue by Orbyt** [react-native-community/react-native-svg#921]
### Description
I'm in the process of swapping some bitmap logos to SVGs for various performance and usability reasons. I am successfully using the the SVG component I built using `react-native-svg` on another screen, however on our "splash" screen a strange issue has become apparent.

### Problem
The logo appears to render at an incorrect position of `0, 0` before it's own parent `<Container/>` renders. Once the `<Container/>` renders, the logo is displayed in the correct position. However, this duplicate render and the change of position causes a flickering effect. This can be seen as the parent `<Container/>` has a bright red background, and the logo can be seen for a short amount of time before the `<Container/>` renders. The issue is reproducible using various SVGs built with the components provided by `react-native-svg`.

Here is the `render` function of the splash screen:

```
<Container style={{ backgroundColor: "red",  alignItems: 'center', flex: 1, justifyContent: 'flex-end' }}>
    <BasketballOutlineIcon size={100} }/>
</Container>
```

The issue is also reproducible when using a simple `<View/>` parent without flexbox properties.

### Video
Below is a video demonstrating the issue. The blue background is the window background, the red background is the `<Container/>` that contains the `<MyLogo/>` component. Notice how the logo renders around position `0, 0` before the red appears.

[Video](https://gfycat.com/thriftytatteredamethystinepython)
 
### Versions

- `"react-native-svg": "9.1.1"`
- `"react": "=16.6.3"`
- `"react-native": "=0.57.8"`
- `"native-base": "^2.10.0"`

**Comment by msand** [comment]
@Orbyt Can you try adding a `return null;` statement in between these two lines:
https://github.com/react-native-community/react-native-svg/blob/b3d29c904508aa8141e678a03049efbee544915c/android/src/main/java/com/horcrux/svg/SvgView.java#L213-L214

**Comment by Orbyt** [comment]
@msand I went into `node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java` and made the following changes:

```java
if (early) {
   ViewParent viewParent = getParent();
   View parent = null;
    if ((viewParent instanceof View)) {
    parent = (View)viewParent;
    }
    float parentWidth = parent == null ? 0 : parent.getWidth();
    float parentHeight = parent == null ? 0 : parent.getHeight();
    width = (float) PropHelper.fromRelative(mbbWidth, parentWidth, 0, mScale, 12);
    height = (float) PropHelper.fromRelative(mbbHeight, parentHeight, 0, mScale, 12);
    setMeasuredDimension((int)Math.ceil(width), (int)Math.ceil(height));
}
```

became:

```java
if (early) {
    return null;
}
```

This appears to fix the flickering icon issue caused by double rendering. Are there any drawbacks to this small fix? I am ignorant of the internal workings of this file; why does this fix work?

**Comment by msand** [comment]
I think this code was in reponse to crashes when width or height had strange values, and animated splash screens with the absolute fill styles, not sure i remember all the details, will try to search for it when I have more time available. But this might be a more reasonable solution either way, having it render for the first time, a few frames later than otherwise, should be an acceptable tradeoff to get use cases without absolute fill to work without flickering/i correct initial positions.

**Comment by Orbyt** [comment]
@msand So what are the next steps here? Is this something I should make a PR for, or would you prefer these changes stay out of the main codebase? We can certainly fork the project and make these changes if needed, but we'd prefer not to maintain our own version of this lib.

**Comment by msand** [comment]
A pull request sounds good. I wonder if there would be need for a call to super.invalidate(); before returning, i have a hunch it might lead to the svg not getting rendered in some edge case. But, I would kind of assume that if the early predicate is true, then the view hierarchy will have at least some layout to do still and will cause another call to draw it later. This is also something that might be okay to release either way, as is, as anyone experiencing issues can use whatever version they had before, and report an issue so we can find a way to deal with any currently unknown/forgotten edge cases. (and hopefully start getting a test suite built)

**Comment by Orbyt** [comment]
@msand Invoking `invalidate()` there would eventually cause `onDraw` to be executed again, which would cause `drawOutput` to be called in a loop until `early` is no longer true. I'm unsure as to whether this is actually an issue; it doesn't appear to have an effect during rendering.

I'll put up a PR this afternoon.

**Comment by msand** [comment]
Fixed in v9.2.4
